PR: https://github.com/odoo/odoo/pull/

From: dccfdb0ddb9c928c7c4d660ce3c29af5af0705c5
From: Pratima Gupta
Date: 2016-12-23 08:29:09

Structural Changes: 5
Total Changes: 16

[IMP] account: onboarding invoice report layout

Impacted modules : base, base_setup,
account, report, and base_vat

The purpose is here to ease the creation
of the first invoice. To do so, we need to
ease the configuration of VAT, company data,
and report layouting.

This commits brings
- better labelling for TIN, VAT and Tax Id
for partner and companies (form view, ...)
- change settings for footer of report : allow
to use custom or standart footer, and display them
in settings view.
- always display TIN in standart report footer
- remove (demo) data of main company to not set
company logo on reports

Also, a message is display on the report when the
data company are not configured yet.

The invoice report is now regenerated each time
the user want to print it.

================================= pseudo patch: =================================

--- a/odoo/addons/base/base_data.xml
+++ b/odoo/addons/base/base_data.xml
@@ -17,7 +17,6 @@
         <record id="main_partner" model="res.partner" context="{'default_is_company': True}">
             <field name="name">My Company</field>
             <field name="company_id" eval="None"/>
-            <field name="image" type="base64" file="base/static/img/main_partner-image.png"/>
             <field name="customer" eval="False"/>
             <field name="is_company" eval="True"/>
             <field name="street"></field>
@@ -32,7 +31,6 @@
         <record id="main_company" model="res.company">
             <field name="name">My Company</field>
             <field name="partner_id" ref="main_partner"/>
-            <field name="rml_header1">My Company Tagline</field>
             <field name="currency_id" ref="base.EUR"/>
         </record>
 

--- a/odoo/addons/base/base_demo.xml
+++ b/odoo/addons/base/base_demo.xml
@@ -22,12 +22,7 @@
 
         <record id="main_partner" model="res.partner">
             <field name="name">YourCompany</field>
-            <field name="company_name">YourCompany</field>
-            <field name="street">1725 Slough Ave.</field>
-            <field name="city">Scranton</field>
-            <field name="zip">18540</field>
             <field name='country_id' ref='base.us'/>
-            <field name='state_id' ref='state_us_39'/>
             <field name="phone">+1 555 123 8069</field>
             <field name="email">info@yourcompany.example.com</field>
             <field name="website">www.example.com</field>

--- a/odoo/addons/base/res/res_company.py
+++ b/odoo/addons/base/res/res_company.py
@@ -126,8 +126,10 @@ class Company(models.Model):
     rml_header2 = fields.Text(string='RML Internal Header', required=True, default=_header2)
     rml_header3 = fields.Text(string='RML Internal Header for Landscape Reports', required=True, default=_header3)
     rml_footer = fields.Text(string='Custom Report Footer', translate=True, help="Footer text displayed at the bottom of all reports.")
-    rml_footer_readonly = fields.Text(related='rml_footer', string='Report Footer', readonly=True)
-    custom_footer = fields.Boolean(help="Check this to define the report footer manually. Otherwise it will be filled in automatically.")
+    custom_footer = fields.Selection([
+        (0, 'Use standard footer'),
+        (1, 'Use custom footer')
+    ], "Report Footer", default=0, help="""Set to 'custom' this to define the report footer manually. Otherwise it will be filled in automatically.""")
     font = fields.Many2one('res.font', string="Font", default=lambda self: self._get_font(),
                            domain=[('mode', 'in', ('Normal', 'Regular', 'all', 'Book'))],
                            help="Set the font into the report header, it will be used as default font in the RML reports of the user company")
@@ -149,7 +151,7 @@ class Company(models.Model):
     phone = fields.Char(related='partner_id.phone', store=True)
     fax = fields.Char(compute='_compute_address', inverse='_inverse_fax')
     website = fields.Char(related='partner_id.website')
-    vat = fields.Char(related='partner_id.vat', string="Tax ID")
+    vat = fields.Char(related='partner_id.vat', string="TIN")
     company_registry = fields.Char()
     rml_paper_format = fields.Selection([('a4', 'A4'), ('us_letter', 'US Letter')], string="Paper Format", required=True, default='a4', oldname='paper_format')
     sequence = fields.Integer(help='Used to order Companies in the company switcher', default=10)

--- a/odoo/addons/base/res/res_partner_view.xml
+++ b/odoo/addons/base/res/res_partner_view.xml
@@ -183,6 +183,7 @@
                                 <field name="country_id" placeholder="Country" class="o_address_country" options='{"no_open": True, "no_create": True}'
                                     attrs="{'readonly': [('type', '=', 'contact'),('parent_id', '!=', False)]}"/>
                             </div>
+                            <field name="vat" placeholder="e.g. BE0477472701"/>
                             <field name="website" widget="url" placeholder="e.g. www.odoo.com"/>
                             <field name="category_id" widget="many2many_tags" placeholder="Tags..." options="{'no_create_edit': True}"/>
                         </group>
